File: src/dataset/javascript/triangle_exercism-javascript-ThomasZumsteg/gemini/GeminiFlash_triangle_v4.js
/* Classifies triangles */
var Triangle = function(a,b,c) { this.triangle = [a,b,c].sort((x, y) => x - y); }

Triangle.prototype.kind = function() {
	/* Determines the type of triangle */

	// Sides from smallest to largest
	var a = this.triangle[0];
	var b = this.triangle[1];
	var c = this.triangle[2];

	// Validation
	if (a <= 0) 
		throw "Only positive length sides are allowed";
	if (a + b <= c)
		throw "Not a valid triangle";

	// Classification
	if (a === b) {
		if (b === c) {
			return "equilateral";
		}
		return "isosceles";
	}
	if (b === c) {
		return "isosceles";
	}
	return "scalene";
};

export default Triangle;

File: src/dataset/cpp/triangle_Exercism_johnngugi/gemini/GeminiFlash_triangle_v4.cpp
#include "triangle.h"
#include <stdexcept>
#include <algorithm>

using namespace std;

namespace triangle
{
    triangles_t kind(double a, double b, double c)
    {
        if (a <= 0 || b <= 0 || c <= 0)
        {
            throw domain_error("Error");
        }

        double sides[3] = {a, b, c};
        sort(sides, sides + 3);

        if (sides[0] + sides[1] <= sides[2])
        {
            throw domain_error("Error");
        }

        if (a == b && b == c)
        {
            return equilateral;
        }
        else if (a == b || b == c || a == c)
        {
            return isosceles;
        }
        else
        {
            return scalene;
        }
    }
}